src/lib/enmail.ts

```diff
@@ -0,0 +1,16 @@
+export interface Message {
+    from?: string;
+    to?: string | string[];
+    content?: string;
+    subject?: string;
+}
+
+export type SendMessage<AdapterOptions = any, SendOptions = any> =
+    (message: Message, adapter: AdapterSend | Adapter<AdapterOptions, SendOptions>, sendOptions?: SendOptions) => Promise<unknown>;
+export type AdapterSend<SendOptions = any> = (message: Message, options?: SendOptions) => Promise<unknown>;
+
+export interface Adapter<AdapterOptions = any, SendOptions = any> {
+    options: AdapterOptions;
+    mailService: any;
```

Comment:
There could be a generic param for mailService. It is almost zero-effort and would help greatly when working directly with the instance as shown [here](https://github.com/AckeeCZ/enmail/pull/8/files#diff-333ca5a62d332aeaa095f662a837dee1R51)

Reply:
`MailService` generic param added